specification/authorization/resource-manager/Microsoft.Authorization/preview/2021-01-01-preview/authorization-RoleAssignmentApprovalCalls.json

```diff
@@ -0,0 +1,844 @@
+          "$ref": "#/definitions/AccessReviewActorIdentity",
+          "description": "Details of the approver.",
+          "x-nullable": true
+        },
+        "reviewedDateTime": {
+          "type": "string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "Date Time when a decision was taken.",
+          "x-nullable": true
+        },
+        "reviewResult": {
+          "type": "string",
+          "description": "The decision on the approval step. This value is initially set to NotReviewed. Approvers can take action of Approve/Deny",
+          "enum": [
+            "Approve",
+            "Deny",
+            "NotReviewed"
+          ],
+          "x-ms-enum": {
+            "name": "RoleAssignmentApprovalStepReviewResult",
+            "modelAsString": true
+          }
+        },
+        "justification": {
+          "type": "string",
+          "description": "Justification provided by approvers for their action"
+        }
+      },
+      "description": "Approval Step."
+    },
+    "RoleAssignmentApprovalStepListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RoleAssignmentApprovalStep"
+          },
+          "description": "Role Assignment Approval Step list."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to use for getting the next set of results."
+        }
+      },
+      "description": "List of role assignment approval step list."
+    },
+    "ErrorDefinition": {
+      "type": "object",
+      "title": "Error",
+      "description": "Error description and code explaining why an operation failed.",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/ErrorDefinitionProperties",
+          "description": "Error of the list gateway status."
+        }
+      }
+    },
+    "ErrorDefinitionProperties": {
+      "type": "object",
+      "title": "Error",
+      "description": "Error description and code explaining why an operation failed.",
+      "properties": {
+        "message": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Description of the error."
+        },
+        "code": {
+          "type": "string",
+          "description": "Error code of list gateway."
+        }
+      }
+    },
+    "Operation": {
+      "description": "The definition of a Microsoft.Authorization operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the operation",
+          "type": "string"
+        },
+        "isDataAction": {
+          "description": "Indicates whether the operation is a data action",
+          "type": "boolean"
+        },
+        "display": {
+          "$ref": "#/definitions/OperationDisplay",
+          "description": "Display of the operation"
+        },
+        "origin": {
```

Comment:
Nit - Origin has set of possible values. Consider making it an enum. [https://github.com/Azure/azure-resource-manager-rpc/blob/eec9f32c0357d0421be13c82ced767943013901a/v1.0/proxy-api-reference.md#origin-details](https://github.com/Azure/azure-resource-manager-rpc/blob/eec9f32c0357d0421be13c82ced767943013901a/v1.0/proxy-api-reference.md#origin-details)

Reply:
Thanks for bringing this up! I've documented all possible values

Reply:
Should model it using x-ms-enum. Take a look at [https://github.com/Azure/azure-rest-api-specs/blob/1907b81ca83cb8b8ad9438284b3f453d73aa3838/specification/common-types/resource-management/v2/types.json#L378](https://github.com/Azure/azure-rest-api-specs/blob/1907b81ca83cb8b8ad9438284b3f453d73aa3838/specification/common-types/resource-management/v2/types.json#L378).

---
In reply to: [551656722](https://github.com/Azure/azure-rest-api-specs/pull/12108#discussion_r551656722) [](ancestors = 551656722)